src/app/sessions/auth/http-auth-injector.coffee

```diff
@@ -11,10 +11,12 @@ angular.module("doubtfire.sessions.auth.http-auth-injector", [])
     injectAuthForRequest = (request) ->
       # Intercept API requests and inject the auth token.
       if _.startsWith(request.url, DoubtfireConstants.API_URL) and currentUser.authenticationToken?
-        request.params = {} unless _.has request, "params"
-        request.params.auth_token = currentUser.authenticationToken
+        request.headers = {'auth_token'} unless _.has request, "headers"
+        request.headers.auth_token = currentUser.authenticationToken
+        request.headers.username = currentUser.profile.username
```

Comment:
This is the right idea, but should be in the same format as the authentication interceptor.